python/tests/test_combinatorics.py

```diff
@@ -976,3 +977,30 @@ def test_forward_time_wright_fisher_simplified(self):
         ts = tables.tree_sequence()
         samples = ts.samples()
         self.verify_topologies(ts, sample_sets=[samples[:10], samples[10:]])
+
+
+class TestTreeNode:
+    """
+    Tests for the TreeNode class used to build simple trees in memory.
+    """
+
+    @pytest.mark.parametrize("n", [2, 3, 5, 10])
+    def test_random_binary_tree(self, n):
```

Comment:
Maybe add a comment that we are checking for a valid binary tree on n leaves, and not checking anything about the randomness

Reply:
Done, good call.